vortex-layout/src/layouts/chunked/reader.rs

```diff
@@ -68,14 +69,21 @@ impl ChunkedReader {
     }
 
     fn chunk_offset(&self, idx: usize) -> u64 {
-        self.chunk_offsets[idx]
+        self.chunk_offsets.get(idx).copied().unwrap_or_else(|| {
+            vortex_panic!(
+                "Internal error: Chunk offset {idx} out of bounds (num_children: {}, num_offsets: {}). \
+                This indicates a bug in ChunkedReader initialization or chunk_range calculation.",
+                self.layout.nchildren(),
+                self.chunk_offsets.len()
+            )
+        })
```

Comment:
This feel a bit silly to produce this panic here explicitly. But we can revisit this in future